src/Transports.Subscriptions.WebSockets/WebSocketReaderPipeline.cs

```diff
@@ -42,26 +42,28 @@ public void LinkTo(ITargetBlock<OperationMessage> target)
 
         public async Task Complete(WebSocketCloseStatus closeStatus, string statusDescription)
         {
-            if (_socket.State != WebSocketState.Closed &&
-                _socket.State != WebSocketState.CloseSent &&
-                _socket.State != WebSocketState.Aborted)
-                try
+            try
+            {
+                if (_socket.State != WebSocketState.Closed &&
+                    _socket.State != WebSocketState.CloseSent &&
+                    _socket.State != WebSocketState.Aborted)
                 {
                     if (closeStatus == WebSocketCloseStatus.NormalClosure)
                         await _socket.CloseAsync(
-                          closeStatus,
-                          statusDescription,
-                          CancellationToken.None);
+                            closeStatus,
+                            statusDescription,
+                            CancellationToken.None);
                     else
                         await _socket.CloseOutputAsync(
-                          closeStatus,
-                          statusDescription,
-                          CancellationToken.None);
-                }
-                finally
-                {
-                    _startBlock.Complete();
+                            closeStatus,
+                            statusDescription,
+                            CancellationToken.None);
                 }
+            }
+            finally
```

Comment:
Basically, the only change here is to move the if condition inside the try block, so the finally block executes regardless of the websocket state.

Reply:
(The finally block terminating the dataflow)